### ISSUE TITLE
Upload pdf / MIME type validation fails

### ISSUE THREAD

**Issue by krisak** [getgrav/grav-plugin-form#88]
I get :
File "koma_plakat_03.pdf" is not an accepted MIME type.

Inside markdown file:
`        - name: file
          label: Soutěžní návrh (náhled v PDF)
          placeholder: Nahrajte návrh
          outerclasses: 'margin'
          type: file
          multiple: false
          destination: "user/data/contact/"
          avoid_overwriting: true
          accept: 'application/pdf'
          validate:
            required: true
`
The select file window makes right asumption and lets me select pdf files (on osx) but the validation fails.

When accept is changed to accept: '*' pdf files work.

**Comment by krisak** [comment]
```
accept: application/pdf
```
Sets the upload select window correclty to expect application/pdf. It does upload without error but the file never shows up (for change).

**Comment by flaviocopes** [comment]
You mean the file is not stored to the user/data/contact folder? is it writeable? Also, try removing the last slash. (`destination: "user/data/contact"`)

**Comment by krisak** [comment]
Yes 100% sure its writable and that other filetypes work and upload correctly. I also tried two servers apache with php 5.6 on localhost and production server running php7 on nginx. Happening on both.

I also tried all kinds of combinations of the accept field - its not realy clear if i should use 
```
accept: 
    - application/pdf 
```
or
```
accept: application/pdf
```
or
```
accept:  'application/pdf'
```
or
```
accept: ['application/pdf']
```

Suprisingly all of them seem to work (upload window expects proper file type) but none solved the pdf check issue.

I will make simple test project to test separately and see what happens.

**Comment by krisak** [comment]
Okey with fresh install and simple form the application/pdf check always fail. I think its a bug. Other filetypes work.

**Comment by flaviocopes** [comment]
Still not seeing an issue.

Did you try with more than one PDF file, or are you trying with a single PDF file that might be failing the check for some reason?

**Comment by flaviocopes** [comment]
Also, all up to date with grav + form plugin updates?

I suppose you're trying in a frontend form, right?

**Comment by krisak** [comment]
Its is frontend form yes. 

I ended up doing my own solution. 

But to the issue - if you just download grav+admin and on the default page try to add form with upload field of pdf. It just wont take any file. I tried many pdf files. I also tried few server configurations. 

The best i could get was when i did accept: '*' and then pdf upload worked on smaller files but it failed on files 3MB+. Actualy it never failed but the files never got to the server. 

I made test repo here: 
https://github.com/krisak/grav-pdf-upload-test

**Comment by flaviocopes** [comment]
I cloned that repo and tested, works just fine to me, out of the box. 
I just tried with MAMP. 

Which environment did you try it on?

**Comment by krisak** [comment]
Well it worked totaly unreliably. Tried on local with AMPPS which is apache 5.6. on production server there was php 7 behind nginx. Upload sizes / memory and everything i can think of was set very high. Image filetypes seemed OK just PDF upload. 

I needed quick form so i made it in wordpress and worked without any problem with exactly same servers and settings. I tried to give you some testcase but if it works for you well than thats very strange. It does not for me in fact it doesnt work on regular hosting.

I cant say what is wrong but from my experience i tried everything.
